### PR TITLE
WIP accept manifest instead of path

### LINKED ISSUE #3: Accept a manifest instead of a path
At the moment, my only option to invoke this is:

```bash
pip-abandoned search .venv/lib/python3.10/site-packages
```

It would be nice if I could also call:

```bash
pip-abandoned search -r requirements.txt
```

(inspiration [pip-audit](https://github.com/pypa/pip-audit) )

I think `requirements.txt` makes the most sense as a starting point for this as users of other tools can usually generate one e.g: `poetry export --format=requirements.txt`, or whatever.

I think the way it makes most sense to do this is:

- Create a temp venv somewhere
- `pip install` into it
- run `search()` on the path to the venv
- Clean up

Some tradeoffs and considerations:

- This will be slower than searching an already installed venv
- Installing everything is wasteful when we only need the metadata: Is there a quicker way to bulk fetch metadata without installing the packages?
- Installing everything will mean we resolve transitive packages
- Installing everything means you don't have to change the exiting code much
- `pip` does not expose a stable library interface
- Should you explicitly require a version of pip?

Needs a bit of thought.

### PR DESCRIPTION
**Pull request by chris48s**
Closes #3

TODO:

- [x] ~more ad-hoc testing: argparse config, multiple requirements files, edge cases~
- [x] ~write tests for `search_files()`~
- [x] ~review terminology for consistency~
- [x] ~documentation updates, help, changelog~

### CODE REVIEW

File: pip_abandoned/lib.py [hunk 1]
```diff
@@ -274,3 +297,16 @@ def search(gh_token, path, verbosity, format_="text"):
...
     return 9
+
+
+def search_files(gh_token, requirements, verbosity, format_="text"):
+    with TemporaryDirectory() as tempdir:
+        site_packages = create_temp_virtualenv(tempdir)
+
+        for reqs in requirements:
+            subprocess.run(
+                [Path(tempdir) / "bin" / "pip", "install", "-r", reqs.absolute()],
```
Comment (chris48s) [inline pip_abandoned/lib.py:308]: instead of looping over the files here, pass multiple `-r` arguments to pip and defer the resolution strategy to pip
